Ramone/cobalt (#59)
HorizenOfficial/evm-documentation: docs/02-connect/01-get_a_wallet.md

To interact with smart contracts in the EVM, you need a valid EVM-compatible wallet. Use one of the following wallets to connect to the EON network:

> * [Cobalt](https://chrome.google.com/webstore/detail/cobalt/hekbjgfncacdinlajhgiakpaieajpfph/related)

Reviewer: Yes, let's omit `/related`

In a web browser, open the Cobalt web extension in the Chrome Web Store. Add the Cobalt extension. Then, click the Cobalt icon in the top-right corner of your browser.  The Cobalt Welcome! page appears.

<p>
<img src="/img/docs/get-started/cobalteon-welcome.png" alt="Cobalt welcome" width="300" height="200" />
</p>

### Create an Account
To begin, create an account for your wallet. Make sure that your password meets the following criteria:

* Your password must contain at least 16-characters.
* No common passwords are accepted, such as password, 12345, etc.
* Your password cannot be saved to a keychain, therefore keep your password in a safe place.

Perform the following steps:

1. Enter a **password**. 
2. Enter your password again to **confirm**.
3. Click **Terms of Use**. After reading the terms and conditions, click the **checkbox** to agree on the terms.
4. Click **Create**.

### Create a New Wallet

1. To create the new wallet, click **Create Wallet**.

<p>
<img src="/img/docs/get-started/gobie-cobalteon-getstarted.png" alt="Cobalt 2-sidechains" width="300" height="200" />
</p>

> 2. The **Choose a wallet** page appears. Click **Horizen EON** or **TokenMint** to create a wallet on a supported sidechain. 

Reviewer: We should direct the user to click "Horizen EON" bc this is EON documentation.
Something like:
```suggestion
2. The **Choose a wallet** page appears. There are options to create a wallet for supported sidechains **Horizen EON** or **TokenMint**. Click **Horizen EON**.
The wallet created for Horizen EON is compatible with any other EVM-compatible chain. 

```

<p>
<img src="/img/docs/get-started/cobalteon-choose-wallet.png" alt="Cobalt 2-sidechains" width="300" height="200" />
</p>

3. The **Create wallet for EON Testnet** page appears. Enter your wallet’s name. Click **Next**. 

<p>
<img src="/img/docs/get-started/cobalteon-wallet-name.png" alt="Cobalt WalletName" width="300" height="200" />
</p>

<p>
<img src="/img/docs/get-started/cobalteon-wallet-name.png" alt="Cobalt Wallet Name" width="300" height="200" />

> </p>

Author: Deleted second image

4. The **Seed Phrase** page appears. Write down your seed phrase. Click **Next**.

<p>
<img src="/img/docs/get-started/gobie-cobalteon-FriedSard-seed.png" alt="Cobalt Wallet Name" width="300" height="200" />
</p>

5. The **Confirm seed phrase** page appears. Select the word in the correct order to confirm your seed phrase. 

<p>
<img src="/img/docs/get-started/gobie-cobalteon-FSseed-confirm.png" alt="Cobalt SeedConfirm" width="300" height="200" />
</p>

6. Click **Confirm**. Cobalt displays your wallet name.

<p>
<img src="/img/docs/get-started/gobie-cobalteon-wallet-complete.png" alt="Cobalt Wallet complete" width="300" height="200" />
</p>

### Import a Wallet

> To restore a previously created wallet, use the **Import Wallet** option. Before you restore a wallet, you need the wallet’s name, password, and seed phrase. 

Reviewer: I would add at the beginning "Alternatively, you can import a wallet."

Author: Made the changed - "Alternatively, you can restore a previously created wallet by using the Import Wallet option."

After installing Cobalt, perform the following steps to import a wallet:

1. In the **Get Started** page, click **Import Wallet**.

<p>
<img src="/img/docs/get-started/cobalteon-import-wallet.png" alt="Cobalt Import Wallet" width="300" height="200" />
</p>

2. The **Choose a wallet** page appears. Select either **Horizen EON** or **TokenMint** for your public sidechain. 

<p>
<img src="/img/docs/get-started/cobalteon-importwallet-choose.png" alt="Cobalt Choose Sidechain" width="300" height="200" />
</p>

> 3. The **Importing wallet for TokenMint Chain** page appears. Enter the wallet name and password. Click **Next**. 

Reviewer: Let's show the flow when clicking "Horizen EON" bc this is EON documentation.
In that case, it will be a 12 word seed phrase
Please also update the screenshot

Author: Updated the flow to show Import Wallet for Horizen Gobi Testnet with new screenshots

Reviewer 2: @Ramone-HL, can you also update the next step (step 4) where it says "Enter your wallet’s 24-words seed phrase." to "Enter your wallet’s 12-words seed phrase." to reflect that of Horizen EON's requirement?
 
<img width="747" alt="Screenshot 2023-04-26 at 1 17 42 PM" src="https://user-images.githubusercontent.com/5024180/234653029-a77c6b2d-77e7-43c3-8431-ce49db7ffb7c.png">

<p>
<img src="/img/docs/get-started/cobalteon-importwallet-seed1.png" alt="Cobalt Wallet Seed" width="300" height="200" />
</p>

5. When the import is complete, the Cobalt wallet displays the restored wallet.

<p>
<img src="/img/docs/get-started/cobalteon-importwallet-complete.png" alt="Cobalt Wallet Complete" width="300" height="200" />
</p>

### Account Settings

The Cobalt wallet provides a setting feature where you can manage your account. 

Click on the the **Account icon** on the top-right corner of the page to display the Account menu:

<p>
<img src="/img/docs/get-started/gobie-cobalteon-account.png" alt="Cobalt Account" width="300" height="200" />
</p>

To manage your Cobalt wallet, use the Account settings to perform the following options:

* **Security** - view your private key or recovery seed phrase
* **Lock** - lock your wallet
* **Connect Sites** - display a list of connected sites
* **Manage Accounts** - rename your wallet name, hide your wallet from view, or add a new account
* **Manage Networks** - delete networks that you configured or add a custom network
* **Terms of Use** - display the Terms of Use page

#### Security

> The Cobalt wallet allows you to view and copy your wallet’s private key. You need the wallet’s private key if you change your web browser or computer. You can also view the recovery seed phrase of your wallet. 

Reviewer: ```suggestion
The Cobalt wallet allows you to view your wallet’s private key. You need the wallet’s private key if you change your web browser or computer. You can also view the recovery seed phrase of your wallet. 

```

1. In the **Account** menu, click **Security**. The **Security** page appears. 

<p>
<img src="/img/docs/get-started/cobalteon-account-security.png" alt="Cobalt Security" width="300" height="200" />
</p>

##### View Private Key

2. Click **View Private Key**. The private key is used to access your account. It is highly recommended that you write down your private key and keep it in a safe place. Enter your password to access your private key. Click **Next**.

<p>
<img src="/img/docs/get-started/gobie-cobalteon-account-privatekey.png" alt="Cobalt PrivateKey" width="300" height="200" />
</p>

##### View Recovery Seed Phrase

3. In the **Security** page, click **View Recovery Seed Phrase**. You will need your wallet’s seed phrase if you change browser or computer. The **View Recovery Seed Phase** page appears. 

<p>
<img src="/img/docs/get-started/cobalteon-account-recovery.png" alt="Cobalt PrivateKey" width="300" height="200" />
</p>

4. The **View Recovery Seed Phase** page displays your secret recovery phrase. Write it down and keep it safe. Click **Close**.

<p>

> <img src="/img/docs/get-started/cobalteon-account-seedphrase.png" alt="Cobalt SeedPhrase" width="300" height="200" />

Reviewer: If possible, I would blur the seed phrase in all screenshots

Author: I tried to blur the seed phrase section of the screenshot with the image tool online but did not work since the tool is limited in capability...it only blurs out the entire screenshot and not selected portions. I will continue to research other tools that can provide blurring selected portions of an image.

</p>

#### Lock 
The Cobalt wallet allows you to lock and unlock your wallet. Use the lock option for security reasons. 

1. In the **Account** menu, click **Lock**. Your wallet is locked. The **Welcome Back!** page appears.

<p>
<img src="/img/docs/get-started/cobalteon-account-welcomeback.png" alt="Cobalt WelcomeBack" width="300" height="200" />
</p>

2. To unlock, enter your password.  Click **Unlock**.

#### Connected Sites

> The Cobalt wallet displays the network addresses that are connected. The Connected Sites page is populated with all site URLs connected to your account, automatically. You can disconnect to a network address if desired. 

Reviewer 2: should we bold **Connected Sites** to distinguish that it is a page?

```suggestion
The Cobalt wallet displays the network addresses that are connected. The **Connected Sites** page is populated with all site URLs connected to your account, automatically. You can disconnect to a network address if desired. 

```

1. In the **Account** menu, click **Connected Sites**. A list of connected network addresses is displayed.

<p>
<img src="/img/docs/get-started/cobalteon-connect-site.png" alt="Cobalt ConnectSite" width="300" height="200" />
</p>

> 2. If desired, you can disconnect from a site by hovering your cursor over the chain icon, where a **Disconnect button** appears. Click the **Disconnect icon** to delete the address.

Reviewer 2: should we rename **Disconnect icon** to **Disconnect button** for consistency?

```suggestion
2. If desired, you can disconnect from a site by hovering your cursor over the chain icon, where a **Disconnect button** appears. Click the **Disconnect button** to delete the address.

```

<p>
<img src="/img/docs/get-started/cobalteon-connectsite-disconnect.png" alt="Cobalt DisConnectSite" width="300" height="200" />
</p>

#### Manage Account

You can manage your wallet accounts. The Manage Account option allows you to rename your account or hide your account from being displayed in Cobalt. This option also allows you to add a new account.

1. In the **Account** menu, click **Manage Account**. A list of valid wallet names and associated addresses are displayed.

<p>
<img src="/img/docs/get-started/gobie-cobalteon-manageacct-list.png" alt="Cobalt ManageAcct List" width="300" height="200" />
</p>

> 2. Click the **pencil icon** next to the wallet name you want to modify. The Rename page appears. In the Account name field, make the changes to the wallet  name. Click **Confirm**.

Reviewer 2: just a couple of tweaks to the icon name and page title:

```suggestion
2. Click the **Pencil icon** next to the wallet name you want to modify. The **Rename** page appears. In the Account name field, make the changes to the wallet  name. Click **Confirm**.

```

<p>
<img src="/img/docs/get-started/gobie-cobalteon-manageacct-rename.png" alt="Cobalt ManageAcct Rename" width="300" height="200" />
</p>

> 3. Toggle the **eye icon** next to the wallet name you want to be visible or not. This option does not delete the account but rather hides the account in the setting option.

Reviewer 2: should we capitalize this to make it consistent with the others?

```suggestion
3. Toggle the **Eye icon** next to the wallet name you want to be visible or not. This option does not delete the account but rather hides the account in the setting option.

```

<p>
<img src="/img/docs/get-started/gobie-cobalteon-manageacct-visible.png" alt="Cobalt ManageAcct Visible" width="300" height="200" />
</p>

4. You can also add an account to your wallet. In the **Manage Account** page, click **Add account**.

   **Note:** When adding a new account, the account’s seed phrase is the same as its initial phrase.

<p>

> <img src="/img/docs/get-started/gobie-manageacct-newacct.png" alt="Cobalt ManageAcct NewAcct" width="300" height="200" />

Reviewer: There is a tiny "g" at the top right corner of the image. Can w crop it?

Author: replaced screenshot with a new one

1. In the **Assets** page, click **Import token**.

<p>
<img src="/img/docs/get-started/cobalteon-import-matic.png" alt="Cobalt ImportToken Matic" width="300" height="200" />
</p>

2. The **Import token** page appears. Enter the **contract address** and click **Import**.

<p>
<img src="/img/docs/get-started/cobalteon-import-token-address.png" alt="Cobalt ImportToken Address" width="300" height="200" />
</p>

3. The **Import token** page displays the import token details. Review the token information. Click **Import**.

* **Contract Address** - The address hosting the contract on the chain that executes functions.
* **Token Symbol** - The name of the token for coin (currency).
* **Token Decimals** - The amount of decimals a token has (divisible from 0 to 18)
* **Balance** - The amount of tokens owned in the selected network.

<p>
<img src="/img/docs/get-started/cobalteon-import-proceed.png" alt="Cobalt ImportToken Proceed" width="300" height="200" />
</p>

> 4. Once the import transaction is complete, the Cobalt wallet displays the imported token in the Assets page.

Reviewer 2: should we bold **Assets** to distinguish that it is a page?

```suggestion
4. Once the import transaction is complete, the Cobalt wallet displays the imported token in the **Assets** page.

```

<p>
<img src="/img/docs/get-started/cobalteon-import-complete.png" alt="Cobalt ImportToken Complete" width="300" height="200" />
</p>

### Send ZEN

The Cobalt wallet allows you to send or transfer ZEN as well as other native tokens to wallets in their respective networks. 

Before sending a transaction, make sure that the appropriate network for recipient address is set in your Cobalt wallet.

To send ZEN, perform the following steps:

> 1. In the **Assets** page, click on the **Asset row**. 

Reviewer 2: I'm not sure users would automatically know what **Asset row** is in this case. should we expound this or update the image to highlight `TZEN 0.51` to clearly identify what **Asset row** is?

Author: Added to text to clarify the meaning of Assets row by using example

<p>
<img src="/img/docs/get-started/gobie-cobalteon-assets.png" alt="Cobalt SendZen Assets" width="300" height="200" />
</p>

2. The **Send ZEN** page appears. Enter transaction details in the following fields:

* **Send to** - Enter the wallet address you want to send to,
* **Amount (ZEN)** - Enter the amount of ZEN you want to send. You can send an equal or less amount displayed in your available balance.  
* **Enter your password** - Enter your wallet’s password, which acts as a signature for this transaction.

3. Click **Next**.

<p>
<img src="/img/docs/get-started/gobie-cobalteon-send-zen.png" alt="Cobalt SendZen" width="300" height="200" />
</p>

4. The **Send ZEN** page appears. Review the address and estimated fee. Click **Send**.

<p>
<img src="/img/docs/get-started/gobie-cobalteon-estimated-fee.png" alt="Cobalt SendZ GasFee" width="300" height="200" />
</p>

5. Review the information and click **Send**. The **Transaction Submitted** page appears. 

<p>
<img src="/img/docs/get-started/gobie-cobalteon-transact-submitted.png" alt="Cobalt SendZ CheckMark" width="300" height="200" />
</p>

> 6. Optionally, verify that the send ZEN transaction is successful. Click **Link in the Explorer** field. The **EON Explorer** appears. It displays the recent Send transaction history.

Reviewer: ```suggestion
6. Optionally, verify that the send ZEN transaction is successful. Click **Link** in the Explorer field. The **EON Explorer** appears. It displays the recent Send transaction history.

```

![cobalt-send-zen-explorer](/img/docs/get-started/gobie-cobalteon-explorer-transdetails.png)

### Send ERC-20 Tokens

> The Cobalt wallet allows sending standard ERC20 tokens to an Ethereum address. 

Reviewer: ```suggestion
The Cobalt wallet allows sending standard ERC-20 tokens to an Ethereum address. 

```

In this example, the LINK token is used for illustrating the send transaction. Make sure you have sufficient gas fees for the transaction. In this case, MATIC is the native currency of the Polygon network.

> To send ERC20 tokens, perform the following steps:

Reviewer: ```suggestion
To send ERC-20 tokens, perform the following steps:

```

1. In the **Assets** page, click on the **Asset row**

<p>
<img src="/img/docs/get-started/cobalteon-send-erc20.png" alt="Cobalt Send ERC20" width="300" height="200" />
</p>

2. The **Send LINK Tokens** page appears. Enter transaction details in the following fields:

* **Send to** - Enter the wallet address you want to send to,
* **Amount (LINK)** - Enter the amount of LINK you want to send. You can send an equal or less amount displayed in your available balance.  
* **Enter your password** - Enter your wallet’s password, which acts as a signature for this transaction.

<p>
<img src="/img/docs/get-started/cobalteon-erc20-link.png" alt="Cobalt Send ERC20 Link" width="300" height="200" />
</p>

> 3. Click **Next**. The **Send Link Tokens** page displays the token amount that is specified, the recipient address, and estimated gas fee for the send transaction. 

Reviewer 2: to be consistent with the image asset like the following:

<img width="682" alt="Screenshot 2023-04-21 at 1 15 10 PM" src="https://user-images.githubusercontent.com/5024180/233696292-8c139f75-0f5a-47b1-ae18-ad863e90054f.png">

should we rename this to **Send LINK Tokens**?

```suggestion
3. Click **Next**. The **Send LINK Tokens** page displays the token amount that is specified, the recipient address, and estimated gas fee for the send transaction. 

```
